src/niraj/narcissisticNumbers.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * Function to check whether a given number is a narcissistic number or not
+ * @param {number} numberToCheck - The number to check
+ * @return {boolean}
+ */
+
+export const isNarcissisticNumber = (numberToCheck: number): boolean => {
+  if (numberToCheck <= 0) return false;
+
+  // All numbers between 1-10 are narcissistic number
+  if (numberToCheck > 0 && numberToCheck < 10) return true;
+
+  const strNum = String(numberToCheck);
+
+  return (
+    numberToCheck ==
+    strNum
+      .split('')
+      .reduce((acc, val) => acc + parseInt(val) ** strNum.length, 0)
+  );
+};
+
+/**
+ * Function to print all the narcissistic numbers between start and end params
+ * @param {number} start - Positive number to start the counting
+ * @param {number} end - Positive number to end the counting
+ * @return {number[]} returns an array of numbers
+ */
+
+export const narcissisticNumbers = (start?: number, end?: number): number[] => {
+  const result = [];
+
+  if (!start || !end || start < 0 || end < 0 || end < start) return result;
+
+  for (let i = start; i <= end; i++) {
```

Comment:
for 'i' also we can define it's type.

Reply:
No necessarily. Since I am defining the variable and immediately assigning the value, typescript can automatically infer that variable's type - I don't need to define the type.